### ISSUE TITLE
Refactor: Make GridSearchTrainer scoring metric configurable

### ISSUE THREAD

**Issue by William-Fernandes252** [William-Fernandes252/loan-default-prediction#6]
## Problem

The `GridSearchTrainer` in `experiments/core/experiment/trainers.py` has a hardcoded scoring strategy:

```python
self._scoring = "balanced_accuracy"
```

This limits flexibility when different experiments require different optimization metrics.

## Recommendation

Make the scoring metric injectable via constructor:

```python
class GridSearchTrainer:
    """Trains models using grid search with injected scoring."""

    def __init__(
        self,
        cv_folds: int = 5,
        scoring: str = "balanced_accuracy",
        n_jobs: int = 1,
    ) -> None:
        self._cv_folds = cv_folds
        self._scoring = scoring
        self._n_jobs = n_jobs
```

This allows creating trainers optimized for different metrics:
- `GridSearchTrainer(scoring="roc_auc")`
- `GridSearchTrainer(scoring="f1")`
- `GridSearchTrainer(scoring="balanced_accuracy")`

## Priority
Low

## Affected Files
- `experiments/core/experiment/trainers.py`
- `experiments/core/experiment/pipeline.py`
